crates/topos-certificate-spammer/src/lib.rs

```diff
@@ -194,14 +195,32 @@ pub async fn run(
     args: CertificateSpammerConfig,
     mut shutdown: mpsc::Receiver<oneshot::Sender<()>>,
 ) -> Result<(), Error> {
-    info!(
-        "Starting topos certificate spammer with the following arguments: {:#?}",
-        args
-    );
-
+    let opts = args.clone();
     // Is list of nodes is specified in the command line use them otherwise use
     // config file provided nodes
-    let target_nodes = if let Some(nodes) = args.target_nodes {
+    let target_nodes = if args.benchmark {
+        if let (Some(hosts), Some(number)) = (args.hosts, args.number) {
+            let uri = hosts
+                .replace("{N}", &0.to_string())
+                .parse::<Uri>()
+                .map_err(|e| Error::BenchmarkConfig(e.to_string()))?;
+
+            if uri.host().is_none() || uri.path().is_empty() || uri.port_u16().is_none() {
+                return Err(Error::BenchmarkConfig(
+                    "Invalid hosts pattern. Has to be in the format of http://validator-1:9090"
+                        .into(),
+                ));
+            }
+
+            (0..number)
+                .map(|n| hosts.replace("{N}", &n.to_string()))
+                .collect::<Vec<String>>()
+        } else {
+            return Err(Error::BenchmarkConfig(
+                "The --benchmark flag needs the following two additional flags being passed to it:\n--hosts http://validator-{N}\n--number 10".into(),
+            ));
+        }
```

Comment:
I don't think you need all those code as you have the `requires` on the `clap` struct

Reply:
But the Rust compiler doesn't know or validate the `requires` flag. We can do an `unwrap()` to get `args.number` for example. 

I would leave this here, because there is no compiler guarantee us removing the `requires` flag at some point.